wayland-client/examples/simple_window.rs

```diff
@@ -111,8 +111,8 @@ impl Dispatch<wl_registry::WlRegistry, ()> for State {
 
 // Ignore events from these object types in this example.
 delegate_noop!(State: wl_compositor::WlCompositor);
-delegate_noop!(State: wl_surface::WlSurface);
-delegate_noop!(State: wl_shm::WlShm);
+delegate_noop!(State: ignore wl_surface::WlSurface);
+delegate_noop!(State: ignore wl_shm::WlShm);
```

Comment:
Wait why are we ignoring WlShm?

I'd think that knowing what formats are supported would be effectively mandatory?

Reply:
> All renderers should support argb8888 and xrgb8888 but any other formats are optional and may not be supported by the particular renderer in use.

https://wayland.app/protocols/wayland

So I think it's perfectly valid for the example to simply assume `wl_shm::Format::Argb8888` is available.